bio-read-quality/main.py

```diff
@@ -0,0 +1,85 @@
+from pathlib import Path
+
+import logging
+
+from Bio import SeqIO
+
+import matplotlib.pyplot as plt
+
+from coretex import currentTaskRun, SequenceDataset, folder_manager
+
+
+def calculateAverageScores(qualityScores: list[list[int]]) -> list[float]:
+    maxLength = max(len(readScores) for readScores in qualityScores)
+
+    totalScores = [0] * maxLength
+    for readScores in qualityScores:
+        for i in range(len(readScores)):
+            totalScores[i] += readScores[i]
+
+    return [score / len(qualityScores) for score in totalScores]
+
+
+def analyzeFastq(sequencePath: Path) -> list[float]:
+    qualityScores: list[list[int]] = []
+    with sequencePath.open("r") as file:
+        for record in SeqIO.parse(file, "fastq"):
+            qualityScores.append(record.letter_annotations["phred_quality"])
+
+    return calculateAverageScores(qualityScores)
+
+
+def createPlot(scores: list[float], forward: bool) -> Path:
+    fig, ax = plt.subplots(figsize = (10, 6))
+    ax.plot(range(len(scores)), scores, linestyle = "-", color = "b", linewidth = 2, label = "Phred Scores")
+
+    if forward:
+        title = "Average Forward Read Quality Scores"
+        plotPath = folder_manager.temp / "forward_qualities.png"
+    else:
+        title = "Average Reverse Read Quality Scores"
+        plotPath = folder_manager.temp / "reverse_qualities.png"
```

Comment:
Make `title` and `plotPath` arguments of `createPlot` function, and remove `forward` as an argument. This way this `if` is no longer needed, because this information is passed in when the function is called.